Tests/SonarScanner.MSBuild.PreProcessor.Test/EngineResolution/EngineResolverTests.cs

```diff
@@ -191,4 +235,24 @@ private void AssertDebugMessages(bool retry, params string[] messages)
         }
         runtime.Logger.DebugMessages.Should().Equal(expected);
     }
+
+    private string TelemetryContent(string telemetryDirectory)
+    {
+        Directory.CreateDirectory(telemetryDirectory);
+        runtime.Logger.WriteTelemetry(telemetryDirectory);
+        var expectedTelemetryLocation = Path.Combine(telemetryDirectory, FileConstants.TelemetryFileName);
+        File.Exists(expectedTelemetryLocation).Should().BeTrue();
+        return File.ReadAllText(expectedTelemetryLocation);
+    }
+
+    // Contents are created with string builder to have the correct line endings for each OS
+    private static string Contents(params string[] telemetryMessages)
+    {
+        var st = new StringBuilder();
+        foreach (var message in telemetryMessages)
+        {
+            st.AppendLine(message);
+        }
+        return st.ToString();
+    }
```

Comment:
Not needed. The TestLogger.TelemetryMessages property should be asserted instead.
```suggestion
```